[게시글 생성/조회/수정/삭제] 에헴 미션 제출합니다.

hyogenie-v opened the pull request:
안녕하세요, 에헴입니다!

스프링 부트는 처음 사용해보지만 강의 덕분에 큰 어려움 없이 미션을 진행할 수 있었습니다.
리뷰 잘 부탁드립니다~

indiveloper committed:
docs README.md 작성

indiveloper committed:
feat : 게시물의 정보를 담고 있는 bean 객체인 article class 구현

indiveloper committed:
docs: README.md 구현 기능 추가

soojinroh committed:
Update README.md

soojinroh committed:
Update README.md

soojinroh committed:
Update README.md

indiveloper committed:
feat 정적 파일 수정 시, 자동 재시작 및 LiveReload 설정

indiveloper committed:
feat 새로운 ArticleId를 생성하는 기능 구현

List<Article>에서 가장 큰 articleId보다 1만큼 큰 Long 타입 값을 반환

indiveloper committed:
fix: Article 클래스 수정

1. Article 클래스를 POJO로 수정
2. 불필요한 필드 삭제

indiveloper committed:
fix: articleId에 해당하는 Article 객체 조회 메서드 수정

indiveloper committed:
feat: 게시글 상세 조회 기능 구현

1. articleId에 해당하는 게시글 조회 메서드 생성
2. 게시글 상세 조회 웹 UI 수정

indiveloper committed:
feat: 게시글 상세 조회 기능 구현

1. articleId에 해당하는 게시글 상세 조회 메서드 구현
2. articleId에 해당하는 게시글을 가져와 Web UI에 구현

indiveloper committed:
feat: 웹 UI 수정

1. index.html, article.html, article-edit.html의 정적 파일 경로 수정
2. 전달받은 데이터를 타임리프를 이용하여 동적으로 출력하는 기능 적용

indiveloper committed:
feat : 게시물 수정 기능 구현

1. 수정하려는 게시글의 정보를 article-edit.html에 출력
2. 전달받은 Article 객체와 동일한 articleId를 가진 Article 객체를 List<Article>에서 바꾸는 메서드 구현
3. coverURL이 입력되지 않은 경우, 기본 이미지 URL 값을 할당하도록 수정

indiveloper committed:
refactor: 정적 파일의 경로를 수정

indiveloper committed:
refactor: 불필요한 파일 삭제

indiveloper committed:
feat : 환경설정 변경

indiveloper committed:
docs : README.md 수정

indiveloper committed:
feat : add lovely sloth image file

live like sloth, eat like sloth, sleep like sloth

indiveloper committed:
refactor : update REST 메서드 변경

1. post 방식의 메서드를 put으로 변경

indiveloper committed:
feat : 게시글 삭제 기능 구현

1. delete 메서드 방식을 이용해 게시글 삭제
2. article.html에 삭제 버튼 추가

indiveloper committed:
feat : 전체 게시글 목록 조회 기능 구현

1. get 메서드 방식을 이용해 전체 게시글 조회 메서드 구현
2. index.html에 전체 글 목록 출력

indiveloper committed:
refactor: css 오류 수정 및 게시글 상세 조회 페이지의 링크 경로 수정

indiveloper committed:
fix : fix conflict

indiveloper committed:
feat: ArticleRepository 테스트 추가

indiveloper committed:
refactor: ArticleController 클래스 테스트 리팩토링

1. 매직 넘버 상수화
2. response의 상태코드 확인 메서드 수정(is3xxRedirection() -> isFound())

indiveloper committed:
refactor : html 중복 제거

hyogenie-v committed:
fix: change port

kang-hyungu reviewed (changes requested):
안녕하세요 에헴님 리뷰어 강현구입니다.
처음이라 스프링 적용이 어려우셨을텐데 코드 잘 작성하셨어요
그런데 로컬 환경에서 index 페이지를 못찾네요. 실패하는 테스트도 있구요.
피드백 확인해보시고 다시 점검 부탁드려요!

hyogenie-v committed:
fix: 그래들 라이브러리 의존성 옵션 변경

hyogenie-v committed:
refactor: ArticleRepository 클래스가 articleId를 부여하는 로직 변경

hyogenie-v committed:
refactor: URI 컨벤션 수정

1. 웹 UI form 태그의 action 경로 수정
2. ArticleController 클래스의 URI 매핑 파라미터가 자원의 행위를 나타내지 않도록 수정
3. 테스트 코드 수정

hyogenie-v committed:
refactor: 중복되는 html 코드 추출 및 include

1. 중복되는 웹 UI의 html 코드를 추출하여 파일을 include하도록 수정한다
2. ArticleController 테스트 클래스에서 상태 코드도 함께 확인하도록 수정한다

hyogenie-v committed:
refactor: 중복 테스트 코드를 setup() 메서드로 추출

hyogenie-v committed:
refactor: 컨트롤러 메서드의 리턴타입 수정

hyogenie-v committed:
refactor: 컨트롤러에서 비즈니스 로직 제거

hyogenie-v committed:
refactor: Article 클래스의 getter 사용 줄임

1. Article 클래스에 hasSameArticleId() 메서드를 추가하여 메시지를 보내도록 수정한다
2. ArticleRepository 클래스의 findArticleIndexByArticle 메서드를 스트림으로 변경한다

hyogenie-v committed:
refactor: 모델 애트리뷰트의 변경 방식을 setter에서 생성자 주입 방식으로 변경

1. 뷰로부터 전달받은 데이터를 ArticleDto 클래스가 받도록 수정한다
2. 수정된 게시글의 데이터를 Article 클래스 내부에서 적용하도록 수정한다

hyogenie-v committed:
refactor: 사용하지 않는 메서드(generateNewId()) 삭제

hyogenie-v committed:
feat: WebTestClient를 사용한 새로운_게시글_추가 API 테스트 추가

kang-hyungu reviewed (changes requested):
에헴님 작업하신 내용 확인했어요
article, articleRepository에 코멘트랑 답변 남겼어요
그리고 ArticleControllerTest에 인수테스트로 변경한게 아직 반영 안된 것 같아요! 요부분도 같이 봐주세요

hyogenie-v committed:
refactor: 인수테스트 수정

1. ArticleControllerTest에서 ArticleRepository 객체를 사용하지 않고 인수테스트를 진행하도록 수정한다
2. ArticleRepository의 메서드를 간결하게 수정한다

hyogenie-v committed:
refactor: 컨트롤러의 Article 객체를 setter가 아닌 생성자로 데이터를 주입받도록 변경

1. Article 클래스에 생성자를 선언한다
2. Article 클래스의 setter를 삭제한다
3. 사용되지 않는 ArticleDto 클래스를 삭제한다
4. 테스트를 수정한다

kang-hyungu reviewed (approved):
피드백 늦어서 죄송해요!
제가 알람을 놓쳤었네요 😭 
수정하신 부분 확인하고 질문 주신 내용에 코멘트 남겼어요
코드 구현하느라 고생하셨어요! 머지 처리할게요
